plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeParquetStatisticsUtils.java

```diff
@@ -110,7 +110,13 @@ public static Object jsonValueToTrinoValue(Type type, @Nullable Object jsonValue
             return (long) (int) jsonValue;
         }
         if (type == BIGINT) {
-            return (long) (int) jsonValue;
+            if (jsonValue instanceof Long) {
+                return (long) jsonValue;
+            }
+            if (jsonValue instanceof Integer) {
+                return (long) (int) jsonValue;
+            }
```

Comment:
I tried out locally (snippet is used in `io.trino.plugin.deltalake.transactionlog.statistics.DeltaLakeJsonFileStatistics`)

```
parseJson(OBJECT_MAPPER, "{\"numRecords\":1,\"minValues\":{\"col\":0},\"maxValues\":{\"col\":9223372036854775807324},\"nullCount\":{\"col\":0}}", DeltaLakeJsonFileStatistics.class)
```

and the parsed value has `BigInteger` . 

However, in this case, I agree  that an exception should be thrown because the value provided is greater than `bigint` max value in Trino https://trino.io/docs/current/language/types.html#bigint